File: scripts/algorithms/Longest Uncommon Subsequence II/Longest Uncommon Subsequence II.py
class Solution:
    def isSub(self, s1, s2):
        n = len(s1)
        i = 0
        for c in s2:
            if i >= n:
                break
            if s1[i] == c:
                i += 1
        return i == n
    
    def findLUSlength(self, strs: List[str]) -> int:
        n = len(strs)
        strs.sort(key = lambda s: -len(s))
        for i in range(n):
            valid = True
            for j in range(n):
                if i != j and self.isSub(strs[i], strs[j]):
                    valid = False
                    break
            if valid:
                return len(strs[i])
        return -1

File: algorithms/L/Longest Uncommon Subsequence II/Longest Uncommon Subsequence II.cpp
class Solution {
private:
    bool isCommon(string &s,string &t){
        int index=0;
        for(int i=0;i<s.size();i++){
            if(s[i]==t[index]){
                if(++index==t.size()){
                    return true;
                }
            }
        }
        return false;
    }
    bool isUncommon(vector<string>&strs,int index){
        for(int i=0;i<strs.size() and strs[index].size()<=strs[i].size();i++){
            if(index!=i and isCommon(strs[i],strs[index])){
                return false;
            }
        }
        return true;
    }
public:
    int findLUSlength(vector<string>& strs) {
        sort(strs.begin(),strs.end(),[](string &s,string &t){
            return s.size()>t.size();
        });
        int ans=-1;
        for(int i=0;i<strs.size();i++){
            if(isUncommon(strs,i)){
                ans=strs[i].size();
                break;
            } 
        }
        return ans;
    }
};